build_runner/lib/src/asset_graph/node.dart

```diff
@@ -27,8 +27,34 @@ abstract class AssetNode {
   /// exist.
   Digest lastKnownDigest;
 
+  /// Whether or not this node was an output of this build.
+  bool get isGenerated => false;
+
+  /// Whether or not this asset can be read.
+  bool get isReadable => true;
+
+  /// Whether or not this node can be used as a primary or secondary input.
```

Comment:
what is a secondary input?

Reply:
any input that isn't primary (something you accessed through `buildStep`)

Reply:
updated this to only mention primary inputs as that was actually the intention behind it